src/main/java/io/jenkins/plugins/jfrog/BinaryInstaller.java

```diff
@@ -80,30 +79,31 @@ public String getId() {
      * @param instance           - JFrogPlatformInstance contains url and credentials needed for the downloading operation
      * @param repository         - Identifies the repository in Artifactory where the CLIs binary is stored
      * @param binaryName         - 'jf' or 'jf.exe'
-     * @param proxyConfiguration - The proxy configuration or null if not configured
+     * @param proxyConfiguration - The proxy configuration
      * @throws IOException in case of any I/O error.
      */
     private static void downloadJfrogCli(File toolLocation, TaskListener log, String providedVersion,
-                                         JFrogPlatformInstance instance, String repository, String binaryName, ProxyConfiguration proxyConfiguration) throws IOException {
+                                         JFrogPlatformInstance instance, String repository, String binaryName, JenkinsProxyConfiguration proxyConfiguration) throws IOException {
         // An empty string indicates the latest version.
         String version = StringUtils.defaultIfBlank(providedVersion, RELEASE);
         String cliUrlSuffix = String.format("/%s/v2-jf/%s/jfrog-cli-%s/%s", repository, version, OsUtils.getOsDetails(), binaryName);
 
         JenkinsBuildInfoLog buildInfoLog = new JenkinsBuildInfoLog(log);
 
         // Downloading binary from Artifactory
-        try (ArtifactoryManager manager = new ArtifactoryManager(instance.inferArtifactoryUrl(), Secret.toString(instance.getCredentialsConfig().getUsername()),
+        String artifactoryUrl = instance.inferArtifactoryUrl();
+        try (ArtifactoryManager manager = new ArtifactoryManager(artifactoryUrl, Secret.toString(instance.getCredentialsConfig().getUsername()),
                 Secret.toString(instance.getCredentialsConfig().getPassword()), Secret.toString(instance.getCredentialsConfig().getAccessToken()), buildInfoLog)) {
-            if (proxyConfiguration != null) {
+            if (proxyConfiguration.isProxyConfigured() && !proxyConfiguration.shouldBypassProxy(artifactoryUrl)) {
```

Comment:
Can we change the `shouldBypassProxy` method, so that it first calls `isProxyConfigured`, and that if the proxy isn't configured, it returns false?
If we do this, we can remove the `isProxyConfigured` check from the above if statement and ny that simplify the code.